randovania/generator/generator.py

```diff
@@ -140,10 +144,20 @@ def _distribute_remaining_items(rng: Random,
     def assign_pickup(node_player: int, node: PickupNode, pickup_target: PickupTarget):
         assignments[node_player].append((node.pickup_index, pickup_target))
 
-    def assign_while_both_non_empty(nodes: list[tuple[int, PickupNode]], pickups: list[PickupTarget]):
+    def shuffle_assignment_lists(nodes: list[tuple[int, PickupNode]], pickups: list[PickupTarget]):
         rng.shuffle(nodes)
         rng.shuffle(pickups)
 
+        # after shuffling, sort both lists such that locations marked as "no majors" come first and major items come
+        # last this will attempt to prevent majors from being assigned to locations marked as "no majors", but allows
+        # for such assignments to happen if there aren't enough remaining locations
```

Comment:
probably better to make this a docstring for the function rather than a series of comments